test/packetimpact/testbench/connections.go

```diff
@@ -157,22 +157,27 @@ func (conn *TCPIPv4) Send(tcp TCP, additionalLayers ...Layer) {
 	if tcp.AckNum == nil {
 		tcp.AckNum = Uint32(uint32(conn.RemoteSeqNum))
 	}
-	layersToSend := deepcopy.Copy(conn.outgoing).(Layers)
-	if err := layersToSend[tcpLayerIndex].(*TCP).merge(tcp); err != nil {
+	layersToSend := deepcopy.Copy(conn.Outgoing).(Layers)
+	if err := layersToSend[tcpLayerIndex].(*TCP).Mergelayer(tcp); err != nil {
 		conn.t.Fatalf("can't merge %v into %v: %s", tcp, layersToSend[tcpLayerIndex], err)
 	}
 	layersToSend = append(layersToSend, additionalLayers...)
-	outBytes, err := layersToSend.toBytes()
+	conn.Sendframe(layersToSend)
+}
+
+// Send a packet with reasonable defaults and override some fields by tcp.
+func (conn *TCPIPv4) Sendframe(Frame Layers) {
+	flags := *Frame[2].(*TCP).Flags
```

Comment:
This could crash because what if `Flags` is nil?

Better to test it for nil and then dereference, like the original code did.

Reply:
[GES] Ok taken care.